src/tools/idlc/src/plugin.c

```diff
@@ -64,6 +72,59 @@ static void *loadsym(void *handle, const char *symbol)
 #endif
 }
 
+#define SUBPROCESS_PIPE_MEMORY_LIMIT 1024 * 1024
+static int run_library_locator(const char *command, char **out_output, int *out_return_code) {
+  size_t output_size = 128;
+  char *output = (char*) malloc(output_size);
+  size_t output_pt = 0;
+  FILE *pipe;
+  int ret = 0;
+  int c;
+
+  if (!output) {
+    // out-of-memory
+    return -1;
+  }
+
+  if ((pipe = SUBPROCESS_POPEN (command, SUBPROCESS_MODE)) == NULL) {
+    free(output);
+    // broken-pipe
+    return -1;
+  }
+
+  while ((c = fgetc(pipe)) != EOF) {
+    if (c == '\n' || c == '\r') {
+      break;
+    }
+
+    output[output_pt++] = (char) c;
+
+    if (output_pt == output_size) {
+      output_size += 128;
+      if (output_size > SUBPROCESS_PIPE_MEMORY_LIMIT) {
+        free(output);
+        SUBPROCESS_PCLOSE (pipe);
+        return -1;
+      }
+
+      char* new = (char*) realloc(output, output_size);
+      if (!new) {
+        free(output);
+        SUBPROCESS_PCLOSE (pipe);
```

Comment:
I wonder if this actually kills the process too. If don't read the output the process might block and I suspect `pclose` (judging by the documentation) simply waits. Maybe good to give that a quick try. The chances of this situation arising is very small of course, but still.

Reply:
I looked into this and you are right that this can cause a hanging process in the way you describe. There is no nice  standard library function that allows us to work around this, by setting a timeout and then killing the subprocess, which is how I would approach it to make it fully 'safe'. I have seen a few solutions, I could implement one but not without significantly increasing the code needed for this PR, effectively doing a full 'subprocess' library again.

Reply:
I figured that would be the case. In PR #842 I dropped a bunch of code that implements exactly this from `ddsrt`. I propose ignoring the issue for now. The chance of this happening is very small and it's not in the core of Cyclone DDS. Eventually, I think we should look into it, but we're starting to maintain a separate abstraction layer for `idlc` (the work I'm doing for #922 also introduces some extra platform specific stuff). I'm going to take a small step back and figure out how/if we can share a bit more code between the two and then revisit this. Would you mind creating an issue so we don't forget?

@eboasson, do you agree with accepting the process isn't killed on timeout?

Reply:
I can learn to stop worrying and love a hanging process 🙂